Documentation/ceph-quickstart.md

```diff
@@ -66,6 +66,52 @@ kind: Namespace
 metadata:
   name: rook-ceph
 ---
+apiVersion: v1
```

Comment:
While all of this is necessary to get started, it doesn't feel like a chunk of yaml this big should go in a quickstart guide. It takes away from the "quick read" nature of the guide. What if we just add a couple of links here that point them to all the details?
- Link to the yaml found at the end of the cluster CRD page
- Link to the example yaml where it is all found in one file

Reply:
hm.... but, without the RBAC stuff, it just doesn't work on more modern clusters (above 1.7?)

Reply:
you mean the linking in addition to the printed file or instead of it? Doing it instead of it, i would be cool with, because then it would not be suggested it'd just work like that.

Reply:
i'm suggesting linking to the full [cluster.yaml](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/cluster.yaml) instead of including it all here. 

The approach with documentation has been so that you could copy/paste into your own yaml files and it should just work, while the same samples are found in the `cluster/examples/kubernetes/ceph` folder so you don't have to copy them into your own yaml file. In this case, the yaml is so big and ugly, even on the detailed [CRD page](https://rook.io/docs/rook/master/ceph-cluster-crd.html#common-cluster-resources), the ugly yaml is saved for the end of the doc.

Reply:
What if we follow the pattern of the [Cluster CRD](https://rook.io/docs/rook/v0.8/ceph-cluster-crd.html) page where we link to the end of the doc for the rest of the yaml? Then it doesn't disrupt the readability of the flow.

Reply:
well... the problem is, that IF you have RBAC enabled in your cluster, the example simply doesn't work. RBAC is default now, so it doesn't work on most proper clusters. 
We could add a sentence to the quickstart that it only works for test cluster like minikube, but i don't know whether that makes it any clearer?

Reply:
Before merge, @jbw976 could you also chime in on this to see if you can think of a better way for the quickstart? I'm still struggling with it, but I'll approve.

Reply:
my opinion here is a bit of a hybrid of the discussion so far.  i agree with @travisn that the quick-start guide should be very easily readable/parseable and all of the RBAC stuff takes away from that.  I also agree with @galexrt and @christianhuening that without the RBAC content then people will miss it and things won't work.

How about a hybrid approach here? Split the cluster RBAC stuff into its own file, e.g. `cluster-rbac.yaml` and then provide as the first step for the user a single line command to install this cluster RBAC stuff, which the user will copy/paste, e.g.:
```
kubectl create -f cluster-rbac.yaml
```
Then the 2nd step would be the full content of JUST the simplified and streamlined cluster CRD yaml.  I really like the idea of the user being able to see the content of the CRD inline in the doc and having it simple and short.  The user can copy-paste that one into their own `cluster.yaml` file which they will run a `kubectl create -f` on, giving them a good opportunity to tweak config, which is also an important scenario.

What do you all think of that hybrid approach? summary:
* break cluster rbac out into its own yaml file
* kubectl create command for cluster RBAC file
* include full content of cluster CRD only in docs

Reply:
I think that sounds like a great idea @jbw976

Reply:
@jbw976 There is a small hurdle, the `cluster.yaml` contains a `rook-ceph` Namespace definition, should be put in both files?
Should we move it out to `namespace.yaml`, put in both cluster and cluster-rbac yamls or use `kubectl apply` instead of `kubectl create`.

What do you think sounds better? I'm personally for moving the namespace out to it's own file.

Reply:
hmm, i would vote to keep the rbac content in the cluster.yaml so we don't add another step to every install. Another idea is to put big comments in the RBAC yaml telling them they can skip over that section in order to focus on the Cluster CRD, which they might need to modify.

Reply:
@travisn Should a bug comment then be in the `cluster.yaml` and the quickstart doc page?

Reply:
To be more clear, I was suggesting to add comments such as this. Think it's helpful or adds too much clutter? 
```
#######################################################
# Define the namespace, service account, and RBAC roles and bindings
# See further below for the cluster CRD
#######################################################
apiVersion: v1
kind: Namespace
...
#######################################################
# The Cluster CRD example
#######################################################
apiVersion: ceph.rook.io/v1beta1
kind: Cluster
...
```

Reply:
@travisn I have added marks like in your comment, PTAL